test/transaction_test.exs

```diff
@@ -98,6 +99,44 @@ defmodule TransactionTest do
     assert query("SELECT 42", []) == [[42]]
   end
 
+  @tag mode: :transaction
+  @tag disconnect_on_read_only_transaction_error: true
+  test "transaction read-only only error disconnects with prepare and execute", context do
+    Process.flag(:trap_exit, true)
+
+    assert transaction(fn conn  ->
+      P.query!(conn, "SET TRANSACTION READ ONLY", []).connection_id
+
+      {:ok, query} = P.prepare(conn, "query_1", "insert into uniques values (1);", [])
+
+      assert capture_log(fn ->
+        {:error, %Postgrex.Error{postgres: %{message: "cannot execute INSERT in a read-only transaction"}}} =
+          P.execute(conn, query, [])
+
+        pid = context[:pid]
+        assert_receive {:EXIT, ^pid, :killed}
+      end) =~ "disconnected: ** (Postgrex.Error) ERROR 25006 (read_only_sql_transaction) cannot execute INSERT"
```

Comment:
We should assert only up to "disconnected: ** (Postgrex.Error) ERROR 25006 (read_only_sql_transaction)" as the rest can be I18ned.